File: get_lower_third/get_lowerthird.py
import json
import boto3
import os
from aws_xray_sdk.core import xray_recorder
from aws_xray_sdk.core import patch_all

xray_recorder.configure(service='Overlay')
plugins = ('EC2Plugin')
xray_recorder.configure(plugins=plugins)
patch_all()
dynamodb = boto3.resource('dynamodb')


def lambda_handler(event, context):
    Index = event["queryStringParameters"]["Index"]
    
    #Get table name from envvars
    TableName = os.environ["lowerthird_table"] if "overlay_table" in os.environ else "lowerthird"
    
    table = dynamodb.Table(TableName)
    
    responce = table.get_item(Key={'Index': Index})
    
    print(responce['Item']['Text'])
    return {
        'statusCode': 200,
        "headers": {
          'Content-Type': 'text/json',
          'Access-Control-Allow-Origin': '*'
        },
        'body': json.dumps(responce['Item']['Text'])
    }

File: get_lower_third/get_lowerthird.go
package main

import (
	"context"
	"errors"
	"fmt"
	"log"
	"os"

	"github.com/aws/aws-lambda-go/lambda"
	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/dynamodb"
	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
)

type Item struct {
	Index string
	Text  string
}
type QueryStringParameters struct {
	Index string `json:"index"`
}
type Request struct {
	QSP QueryStringParameters `json:"queryStringParameters"`
}

func HandleRequest(ctx context.Context, req Request) (string, error) {
	tableName := os.Getenv("lowerthird_table")
	if tableName == "" {
		tableName = "lowerthird"
	}

	sess := session.Must(session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
	}))

	svc := dynamodb.New(sess)

	result, err := svc.GetItem(&dynamodb.GetItemInput{
		TableName: aws.String(tableName),
		Key: map[string]*dynamodb.AttributeValue{
			"Index": {
				S: aws.String("0"),
			},
		},
	})
	if err != nil {
		log.Fatalf("Got error calling getItem: %s", err)
	}
	if result.Item == nil {
		return "", errors.New("Could not find")
	}

	item := Item{}

	err = dynamodbattribute.UnmarshalMap(result.Item, &item)
	if err != nil {
		panic(fmt.Sprintf("Failed to unmarshal Record, %v", err))
	}
	return fmt.Sprintf("%s", item.Text), nil
}

func main() {
	lambda.Start(HandleRequest)
}
